lemminx-liberty/src/main/java/io/openliberty/tools/langserver/lemminx/codeactions/ReplacePlatform.java

```diff
@@ -0,0 +1,126 @@
+                // check for conflicting platforms for already existing platforms. remove them from quick fix items
+                List<String> replacementPlatformsWithoutConflicts = getReplacementPlatformsWithoutConflicts(replacementPlatforms, existingPlatforms);
+                // filter with entered word
+                List<String> filteredPlatforms = replacementPlatformsWithoutConflicts.stream().
+                        filter(it -> it.toLowerCase().contains(platformNameToReplace.toLowerCase()))
+                        .toList();
+                for (String nextPlatform : filteredPlatforms) {
+                    if (!nextPlatform.equals(platformNameToReplace)) {
+                        String title = "Replace platform with " + nextPlatform;
+                        codeActions.add(CodeActionFactory.replace(title, diagnostic.getRange(), nextPlatform, document.getTextDocument(), diagnostic));
+                    }
+                }
+            }
+        } catch (Exception e) {
+            // BadLocationException not expected
+            LOGGER.warning("Could not generate code action for replace platform: " + e);
+        }
+    }
+
+    /**
+     * get list of existing platforms to exclude from list of possible replacements
+     * also exclude any platform with a different version that matches an existing platform
+     * @param allPlatforms all platforms
+     * @return replacement platforms
+     */
+    private static List<String> getReplacementPlatforms(Set<String> allPlatforms, List<String> existingPlatforms) {
+        List<String> existingPlatformsWithoutVersionLowerCase = existingPlatforms.stream().map(p->LibertyUtils.stripVersion(p).toLowerCase()).toList();
+        return allPlatforms.stream().filter(
+                p -> !existingPlatformsWithoutVersionLowerCase.contains(LibertyUtils.stripVersion(p.toLowerCase()))
+        ).sorted(Comparator.naturalOrder()).toList();
+    }
+
+    /**
+     * find and remove conflicting platforms
+     * @param replacementPlatforms replacement platform  list
+     * @param existingPlatforms existing platforms in doc
+     * @return non-conflicting platforms
+     */
+    private static List<String> getReplacementPlatformsWithoutConflicts(List<String> replacementPlatforms, List<String> existingPlatforms) {
+        List<String> replacementPlatformsWithoutConflicts=new ArrayList<>();
+        replacementPlatforms.forEach(
+                p->{
+                    String pWithoutVersion = LibertyUtils.stripVersion(p.toLowerCase());
+
+                    Optional<String> conflictingPlatform = existingPlatforms.stream().filter(
+                            existingPlatform -> {
+                                String conflictingPlatformName = LibertyConstants.conflictingPlatforms.get(pWithoutVersion);
+                                return conflictingPlatformName != null && existingPlatform.toLowerCase().contains(conflictingPlatformName);
```

Comment:
conflictingPlatforms have preserved case and existingPlatforms are lower case, correct? Which means there are scenarios where this check will not work. Try to add another test to make it fail.

Reply:
This one is not a problem since conflictingPlatforms is already all lower case.

Reply:
As discussed, its working fine